### PR TITLE
Onboarding checklist on dashboard

### PR DESCRIPTION
**Pull request by hunar1312**

### CODE REVIEW

File: app/routes/_layout+/dashboard.tsx [hunk 1]
```diff
@@ -106,36 +108,48 @@ export const handle = {
 };
 
 export default function DashboardPage() {
+  const [completedAllChecks] = useState(false);
+  const matches = useMatches();
+  const parentRoutesData = matches.find(
+    (match) => match.id === "routes/_layout+/_layout"
+  )?.data;
+  const { skipOnboardingChecklist } = parentRoutesData;
```
Comment (DonKoko) [inline app/routes/_layout+/dashboard.tsx]: No reason for this to be in the parent route. Should be handled within the dashboard route.

File: app/routes/_layout+/dashboard.tsx [hunk 2]
```diff
@@ -106,36 +108,48 @@ export const handle = {
...
-    <div className="pb-8">
-      <AnnouncementBar />
-      <div className="w-full">
-        <AssetsForEachMonth />
-      </div>
-      <div className="pb-4 xl:flex xl:gap-4">
-        <div className="xl:lg-1/2 w-full">
-          <AssetsByStatusChart />
+    <div>
+      {completedAllChecks && skipOnboardingChecklist ? (
```
Comment (DonKoko) [inline app/routes/_layout+/dashboard.tsx]: This should be an OR

File: app/components/dashboard/checklist.tsx [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+                    </h6>
+                    <p className=" text-gray-600">
+                      Each asset gets it’s own encrypted QR tag.
+                    </p>
+                  </div>
+                  <div className="flex gap-3">
+                    <Link to="#" className=" font-semibold text-gray-600">
+                      Learn more
+                    </Link>
+                    <Button variant="link" to="/categories">
```
Comment (DonKoko) [inline app/components/dashboard/checklist.tsx]: Wrong link

File: app/components/dashboard/checklist.tsx [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+                    <p className=" text-gray-600">
+                      View, edit or delete our default categories and build your
+                      own.
+                    </p>
+                  </div>
+                  <div className="flex gap-3">
+                    <Link to="#" className=" font-semibold text-gray-600">
+                      Learn more
+                    </Link>
+                    <Button variant="link" to="/categories">
```
Comment (DonKoko) [inline app/components/dashboard/checklist.tsx]: Wrong link

File: app/components/dashboard/checklist.tsx [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+                    <p className=" text-gray-600">
+                      Tags are small pieces of information that can be added to
+                      assets.
+                    </p>
+                  </div>
+                  <div className="flex gap-3">
+                    <Link to="#" className=" font-semibold text-gray-600">
+                      Learn more
+                    </Link>
+                    <Button variant="link" to="/tags">
```
Comment (DonKoko) [inline app/components/dashboard/checklist.tsx]: Wrong link

File: app/components/dashboard/checklist.tsx [hunk 6]
```diff
@@ -0,0 +1,251 @@
...
+                    <p className=" text-gray-600">
+                      Track who has custody over an asset by adding your team
+                      members to shelf.
+                    </p>
+                  </div>
+                  <div className="flex gap-3">
+                    <Link to="#" className=" font-semibold text-gray-600">
+                      Learn more
+                    </Link>
+                    <Button variant="link" to="/settings/team">
```
Comment (DonKoko) [inline app/components/dashboard/checklist.tsx:191]: Wrong link

File: app/components/dashboard/checklist.tsx [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+                    </h6>
+                    <p className=" text-gray-600">
+                      Improve your asset database with custom field types.
+                    </p>
+                  </div>
+                  <div className="flex gap-3">
+                    <Link to="#" className=" font-semibold text-gray-600">
+                      Learn more
+                    </Link>
+                    <Button variant="link" to="/settings/custom-fields">
```
Comment (DonKoko) [inline app/components/dashboard/checklist.tsx]: Wrong link

**Review by DonKoko — CHANGES_REQUESTED** [review]
Made some comments and added the function that calculates each checklist item.

**Comment by DonKoko** [comment]
Fixed a few things, you can see here: https://github.com/Shelf-nu/shelf.nu/pull/559/commits/5e7414033b23658526104f661fe8de2ff85362a0

Fallbacks are still not good. They should represent the actual structure of the elements and be the same height. Currently they are too tall, they are the same for each element and they have some profile picture and rows inside of them. The purpose is to prevent layout shift which also happens now.

**Comment by jurrejansen** [comment]
Learn more links|:

Create your first asset: https://www.shelf.nu/knowledge-base/adding-new-assets
Create a custom category: https://www.shelf.nu/knowledge-base/using-categories-to-organize-your-asset-inventory
Create a tag: Need to make post
Add a team member: https://www.shelf.nu/knowledge-base/onboarding-your-team-members
Assign custody over an asset: https://www.shelf.nu/knowledge-base/custody-feature-for-long-term-equipment-lend-outs
Create a custom field: https://www.shelf.nu/knowledge-base/adding-additional-fields-to-assets

**Pull request merged**